ion-schema/src/model/type_definition.rs

```diff
@@ -17,10 +17,16 @@ use std::ops::ControlFlow;
 pub type VersionedTypeDefinition<V> = Versioned<TypeDefinition, V>;
 
 /// A TypeDefinition is a set of constraints and (optionally) additional user content.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone)]
 pub struct TypeDefinition {
     constraints: Bag<AnyConstraint>,
     open_content: Bag<(Symbol, IonData<Element>)>,
+    source_location: IslSourceLocation,
+}
+impl PartialEq for TypeDefinition {
+    fn eq(&self, other: &Self) -> bool {
+        self.constraints == other.constraints && self.open_content == other.open_content
+    }
```

Comment:
❗ As with `Import`, `source_location` is extra metadata in `TypeDefinition`, not actually part of the equality, so we need a custom implementation of `PartialEq`.